typescript/packages/common-charm/src/storage.ts

```diff
@@ -120,12 +137,18 @@ type Job = {
  * is probably already further ahead.
  */
 class StorageImpl implements Storage {
-  constructor(private storageProvider: StorageProvider) {
+  constructor() {
     const [cancel, addCancel] = useCancelGroup();
     this.cancel = cancel;
     this.addCancel = addCancel;
+
+    if (location?.href) this.setRemoteStorage(new URL(location.href));
```

Comment:
using location here means it will only work on a browser - deno doesn't have `location` and so we might need a way to instantiate the storage elsewhere / send the url to the constructor